bin/faucet/src/client.rs

```diff
@@ -69,11 +70,12 @@ impl FaucetClient {
             root_block_header,
             root_chain_mmr,
         );
-        let authenticator = BasicAuthenticator::<StdRng>::new(&[(
+        let authenticator = Arc::new(BasicAuthenticator::<StdRng>::new(&[(
             secret.public_key().into(),
             AuthSecretKey::RpoFalcon512(secret),
-        )]);
-        let executor = TransactionExecutor::new(data_store.clone(), Some(Rc::new(authenticator)));
+        )])) as Arc<dyn TransactionAuthenticator>;
+        let executor =
+            TransactionExecutor::new(Arc::new(data_store.clone()), Some(authenticator.clone()));
```

Comment:
Nit: here is better to refactor a bit (return `Arc<FaucetDataStore>` from the function and just clone `Arc`'s, not data store itself.

Reply:
But we need to provide the `FaucetDataStore` to the constructor at the end of the function, so would it be efficient to wrap the  `FaucetDataStore` into `Arc` just to provide it to the `TransactionExecutor` and then somehow unwrap it back (with `into_inner`?) to provide to the constructor? I understand the downsides of cloning the actual value instead of `Arc`, but here it feels like it brings more complexity, no?

Reply:
No-no, we can't unwrap it from `Arc` after cloning. I meant, we can refactor this in order to use `Arc<FaucetDataStore>` everywhere. :)

Reply:
But okay, just leave `TODO` here, I'm refactoring faucet and will do this by myself later.